scan.go

```diff
@@ -325,21 +373,33 @@ func (s *Scan) closeScan(server *connection, location *RegionInfo, id uint64) {
 		ScannerId:    pb.Uint64(id),
 		CloseScanner: pb.Bool(true),
 	}
+
 	cl := newCall(req)
-	server.call(cl)
+	err := server.call(cl)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	// TODO: add exception check.
 	<-cl.responseCh
+	return nil
 }
 
-func (s *Scan) getServerAndLocation(table, startRow []byte) (server *connection, location *RegionInfo) {
-	if s.server != nil && s.location != nil {
-		server = s.server
-		location = s.location
-		return
+func (s *Scan) getServerAndLocation(table, startRow []byte) (*connection, *RegionInfo, error) {
+	var err error
+	if s.location == nil {
+		s.location, err = s.client.LocateRegion(table, startRow, true)
```

Comment:
You should add 
`s.server, err = s.client.getRegionConn(s.location.Server)`
Because location has been modified and server need re-generate with new location